### DISCUSSION TITLE
which source files communicate with the intelliflo pump?

### DISCUSSION THREAD

**Discussion by kevincw01** [tagyoureit/nodejs-poolController#583] [RS485]
I am interfacing with my pump using a small microcontroller and I was hoping to leverage the source code developed in this project to b ootstrap my development.  I have dug around quite a bit but there's a lot of abstraction and I can't quite put my finger on the specific source file(s) that actually send and receive with the pump.  Any tips would be appreciated.

**Comment by rstrouse** [comment]
Take a look at `/controller/nixie/pumps/Pump.ts`.  Bear in mind there are different pump types classes in there.

**Reply by kevincw01** [reply]
Thx.  Am I correct that the functions in that file that actually communicate with the pump are setDriveStateAsync, requestPumpStatus, setPumpToRemoteControl, setPumpFeature... Etc.  Basically everywhere with a "conn.queueSendMessage(out)"?

**Comment by rstrouse** [comment]
That is correct.  The returns are here https://github.com/tagyoureit/nodejs-poolController/blob/master/controller/comms/messages/status/PumpStateMessage.ts
